client/my-sites/stats/stats-post-anniversaries/index.jsx

```diff
@@ -0,0 +1,243 @@
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import StatsModuleExpand from '../stats-module/expand';
+import SectionHeader from 'components/section-header';
+import PostsList from './posts-list';
+import QueryPosts from 'components/data/query-posts';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePostsForQuery } from 'state/posts/selectors';
+import QueryPostStats from 'components/data/query-post-stats';
+import { getPostStats } from 'state/stats/posts/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+
+// Maximum number of simultaneous queries to fetch the posts by year
+const MAX_POST_QUERIES = 10;
+
+// Maximum number of posts to display
+const MAX_POSTS_DISPLAYED = 3;
+
+const PERIODS_ALLOWED = [ 'day', 'week', 'month' ];
+
+class StatModulePostAnniversaries extends Component {
+	render() {
+		const {
+			topPostsOnly,
+			oldestPostQuery,
+			postsByYearQueries,
+			postsByYear,
+			siteId,
+			translate,
+			summary,
+			period,
+			path,
+			siteSlug,
+		} = this.props;
+
+		if ( ! siteId || PERIODS_ALLOWED.indexOf( period.period ) === -1 ) {
+			return null;
+		}
+
+		const allPosts = Array.prototype.concat( ...postsByYear );
+
+		const title = topPostsOnly
+			? translate( 'Top %d anniversaries', {
+					args: MAX_POSTS_DISPLAYED,
+					comment: 'Title of the anniversaries module (when the maximum is reached)',
+				} )
+			: translate( 'Anniversaries', { comment: 'Title of the anniversaries module' } );
+
+		return (
+			<div>
+				{ <QueryPosts siteId={ siteId } query={ oldestPostQuery } /> }
+				{ postsByYearQueries.map(
+					( query, i ) => <QueryPosts key={ i } siteId={ siteId } query={ query } />,
+				) }
+				{ ! allPosts.length
+					? null
+					: <div>
+							{ topPostsOnly
+								? allPosts.map(
+										( post, i ) => (
+											<QueryPostStats key={ i } siteId={ siteId } postId={ post.ID } />
+										),
+									)
+								: null }
+							{ ! summary &&
+								<SectionHeader label={ title } href={ summary ? null : this.getHref() } /> }
+							<Card className="stats-post-anniversaries__card stats-module">
+								{ summary &&
+									<div className="stats-section-title">
+										<h3>
+											{ translate( 'Anniversaries for %s', { args: this.getDateForDisplay() } ) }
+										</h3>
+									</div> }
+								<PostsList
+									allPosts={ allPosts }
+									postsByYear={ postsByYear }
+									summary={ summary }
+									period={ period }
+								/>
+								{ topPostsOnly && <StatsModuleExpand href={ this.getHref() } /> }
+							</Card>
+						</div> }
+			</div>
+		);
+	}
+
+	getDateForDisplay() {
+		const { period, translate } = this.props;
+
+		// Ensure we have a moment instance here to work with.
+		const date = period.startOf.clone();
+
+		if ( period.period === 'week' ) {
+			return translate( '%(startDate)s - %(endDate)s', {
```

Comment:
Hi! I've found a possible matching string that has already been translated **32** times:
`translate( '%(startDate)s - %(endDate)s', { context: 'Date range for which stats are being displayed'} )` *ES Score: 27*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?